src/main/java/teammates/common/util/Url.java

```diff
@@ -107,34 +65,38 @@ public String get(String parameterName) {
      * </ul>
      */
     public static String addParamToUrl(String url, String key, String value) {
-        if (key == null || value == null) {
-            // return the url if any of the key or the value is null
+        if (key == null || key.isEmpty() || value == null || value.isEmpty()
+             || url.contains("?" + key + "=") || url.contains("&" + key + "=")) {
+            // return the url if any of the key or the value is null or empty
+            // or if the key is already included in the url
             return url;
         }
-        if (url.contains("?" + key + "=") || url.contains("&" + key + "=")) {
-            // return the url if the key is already included in the url
-            return url;
-        }
-        url += url.indexOf('?') >= 0 ? '&' : '?';
-        url += key + "=" + Sanitizer.sanitizeForUri(value);
-        return url;
+        return url + (url.contains("?") ? "&" : "?") + key + "=" + Sanitizer.sanitizeForUri(value);
     }
 
     public static String trimTrailingSlash(String url) {
         return url.trim().replaceAll("/(?=$)", "");
     }
+    
+    public static String getRelativePath(String url) {
+        try {
+            return new Url(url).toString();
+        } catch (AssertionError e) {
+            return url;
+        }
+    }
```

Comment:
1. Did you mean support only Firefox for *testing* right? I assume the difference in behavior affects testing only? In that case we should keep the production code clean and apply any workaround in the test code. e.g. do an if-else based on WebDriver.
2. Yes, I think at some point we stopped using Chrome for testing. There are many other quirks between the two versions that gets in the way of comparing html. We can check if anyone is using Chrome.